specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/virtualNetworks.json

```diff
@@ -0,0 +1,575 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Lists all virtualnetworks under the resource group.",
+            "schema": {
+              "$ref": "#/definitions/virtualnetworksListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AzureStackHCI/virtualnetworks": {
+      "get": {
+        "operationId": "virtualnetworks_ListBySubscription",
+        "x-ms-examples": {
+          "ListVirtualNetworkBySubscription": {
+            "$ref": "./examples/ListVirtualNetworkBySubscription.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Lists all virtualnetworks under the subscription.",
+            "schema": {
+              "$ref": "#/definitions/virtualnetworksListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "virtualnetworksProperties": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "properties": {
+        "networkType": {
+          "type": "string",
+          "description": "Type of the network"
+        },
+        "subnets": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "Name - The name of the resource that is unique within a resource group. This name can be used to access the resource."
+              },
+              "properties": {
+                "type": "object",
+                "x-ms-client-flatten": true,
+                "properties": {
+                  "addressPrefix": {
```

Comment:
String enum

Reply:
As mentioned in a previous comment, we cannot change this without breaking customers for this API (the API was published in RPSaaSDev a few months ago and we forgot to port that API to main). That said, in the long run, I agree with changing this to string enum. We will pick this up in the next version of the API.